source/backend/entities/Helpers/Extensions/TypeCodeExtensions.cs

```diff
@@ -0,0 +1,19 @@
+using System;
+
+namespace Pims.Dal.Entities.Helpers.Extensions
+{
+    public static class TypeCodeExtensions
+    {
+        /// <summary>
+        /// Determines whether a given type is currently expired, based on its EffectiveDate and ExpiryDate values
+        /// </summary>
+        /// <param name="type">The type code.</param>
+        /// <param name="currentDate">The current date.</param>
+        /// <returns>True if the type is expired; false otherwise.</returns>
+        public static bool IsExpiredType(this IExpiringTypeEntity type, DateTime? currentDate = null)
+        {
+            DateTime now = currentDate.HasValue ? (DateTime)currentDate?.Date : DateTime.UtcNow.Date;
+            return (type.EffectiveDate.Date > now) || (type.ExpiryDate.HasValue && type.ExpiryDate?.Date <= now);
```

Comment:
Hmm is > correct or >= for the effective date?

Reply:
The logic is to determine when a type code is expired (or not). Based on conversations for another entity on the system that supports the same concept of "expiry" with two dates (effective - expiryDate) this is the actual text from Confluence:

> **Effective date:** As of this date the code is available for selection in PIMS
> **Expiry date:**  As of this date the code is not available in PIMS for selection.
>
> **NOTE**: Since both effective and expiry dates are inclusive of the provided date, the expiry date should be at least one day after the effective date

So the here the code is expired if the effective date has not been reached yet, but will become active the day of so > makes sense here for effective date. Also remember that we are dealing with dates without time component (by doing UtcNow.Date)